http/src/request/try_into_request.rs

```diff
@@ -0,0 +1,334 @@
+    impl Sealed for UpdateGuildWelcomeScreen<'_> {}
+    impl Sealed for UpdateGuildWidget<'_> {}
+    impl Sealed for UpdateMessage<'_> {}
+    impl Sealed for UpdateOriginalResponse<'_> {}
+    impl Sealed for UpdateRole<'_> {}
+    impl Sealed for UpdateRolePositions<'_> {}
+    impl Sealed for UpdateStageInstance<'_> {}
+    impl Sealed for UpdateTemplate<'_> {}
+    impl Sealed for UpdateThread<'_> {}
+    impl Sealed for UpdateUserVoiceState<'_> {}
+    impl Sealed for UpdateWebhook<'_> {}
+    impl Sealed for UpdateWebhookMessage<'_> {}
+    impl Sealed for UpdateWebhookWithToken<'_> {}
+}
+
+use super::base::Request;
+use crate::error::Error;
+
+/// Convert a typed request builder into a raw [`Request`].
+///
+/// Converting a typed request builder into a raw request may be preferable in
+/// order to verify whether a request is valid prior to passing it to
+/// [`Client::request`].
+///
+/// Creating raw requests is useful for unit tests and debugging.
+///
+/// # Examples
+///
+/// Convert a [`CreateMessage`] builder into a [`Request`], inspect its body
+/// and route, and then send the request:
+///
+/// ```no_run
+/// # #[tokio::main] async fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// use std::{env, str};
+/// use twilight_http::{client::Client, request::TryIntoRequest};
+/// use twilight_model::{channel::Message, id::ChannelId};
+///
+/// let client = Client::new(env::var("DISCORD_TOKEN")?);
+/// let channel_id = ChannelId::new(1).expect("non zero");
+/// let builder = client.create_message(channel_id)
+///     .content("This is a test message!")?
+///     .tts(false);
+///
+/// let request = builder.try_into_request()?;
+///
+/// println!("{:?} {}", request.method(), request.path());
+///
+/// if let Some(body) = request.body() {
+///     println!("{}", str::from_utf8(body)?);
+/// }
+///
+/// // Because a raw request is being performed, the output type must be
+/// // explicit.
+/// let response = client.request::<Message>(request).await?;
+/// # Ok(()) }
+/// ```
+///
+/// [`Client::request`]: crate::client::Client::request
+/// [`CreateMessage`]: super::channel::message::CreateMessage
+pub trait TryIntoRequest: private::Sealed {
+    /// Try to convert a request builder into a raw [`Request`].
+    ///
+    /// # Errors
+    ///
+    /// Not all typed request builder conversions return an error and may
+    /// instead always succeed. Refer to the documentation for each
+    /// implementation for clarification.
+    ///
+    /// Requests may return an error type of [`ErrorType::CreatingHeader`] if
+    /// creating an audit log header value fails.
+    ///
+    /// Requests may return an error type of [`ErrorType::Json`] if serializing
+    /// a request body fails.
+    ///
+    /// [`ErrorType::CreatingHeader`]: crate::error::ErrorType::CreatingHeader
+    /// [`ErrorType::Json`]: crate::error::ErrorType::Json
+    fn try_into_request(self) -> Result<Request, Error>;
```

Comment:
By making the error generic implementations that never error can opt for [Infallible](https://doc.rust-lang.org/std/convert/enum.Infallible.html) (or [never](https://doc.rust-lang.org/std/primitive.never.html) once it's stable) thus clearly declaring that it never errors.

```suggestion
pub trait TryIntoRequest<E = Error>: private::Sealed {
    /// Try to convert a request builder into a raw [`Request`].
    ///
    /// # Errors
    ///
    /// Requests may return an error type of [`ErrorType::CreatingHeader`] if
    /// creating an audit log header value fails.
    ///
    /// Requests may return an error type of [`ErrorType::Json`] if serializing
    /// a request body fails.
    ///
    /// [`ErrorType::CreatingHeader`]: crate::error::ErrorType::CreatingHeader
    /// [`ErrorType::Json`]: crate::error::ErrorType::Json
    fn try_into_request(&self) -> Result<Request, E>;
```

Reply:
Won't move forward with this since it locks us in too much in how the API can respond to Discord's changes